core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -650,6 +656,19 @@ private[spark] class BlockManager(
       blockId: BlockId,
       level: StorageLevel,
       classTag: ClassTag[_]): StreamCallbackWithID = {
+    // Delegate shuffle blocks here to resolver if supported
+    if (blockId.isShuffle || blockId.isInternalShuffle) {
+      logDebug(s"Putting shuffle block ${blockId}")
+      try {
+        return indexShuffleResolver.putShuffleBlockAsStream(blockId, serializerManager)
+      } catch {
+        case e: ClassCastException => throw new Exception(
+          s"Unexpected shuffle block ${blockId} with unsupported shuffle " +
+          s"resolver ${shuffleManager.shuffleBlockResolver}")
+      }
+    }
+    logDebug(s"Putting regular block ${blockId}")
+    // All other blocks
```

Comment:
Can a blockId be representing a shuffle block and also a regular block at same time?

Reply:
No, look at the blockId case classes.

Reply:
Then here we should have `else` block?

Reply:
We return inside of the if block so if we've reached here it is a not a shuffle block.

Reply:
ah, oops, missed it.